src/slicegpt/model_adapter.py

```diff
@@ -293,6 +297,123 @@ def convert_layer_to_compressed_and_register_buffers(self, layer: Module, layer_
         compressed_layer.register_parameter('attn_shortcut_Q', None)
         return compressed_layer
 
+    def post_init(self, tokenizer: PreTrainedTokenizerBase) -> None:
+        """
+        This method is called after the model is initialized and all the properties are set.
+        Override in subclasses to perform any additional setup.
+        """
+        pass
+
+    @classmethod
+    def from_model(
+        cls,
+        model_name: str,
+        model_path: str,
+        *,
+        model_type: str = 'pretrained',
+        dtype: torch.dtype = torch.float16,
+        local_files_only: bool = False,
+        token: str | bool | None = None,
+    ) -> ModelAdapter:
+        """
+        Load the model from the given path and return the corresponding ModelAdapter instance.
+        `model_type` can be 'pretrained' or 'uninitialized'.
```

Comment:
It should also be mentioned, that for this method to work the correct adapter needs to be imported in advance (otherwise, it won't be present in the subclass hierarchy).

Reply:
Also applies to docstrings for hf_utils methods.

Reply:
Added a note in the relevant places.